src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -281,12 +281,13 @@ protected function getConstructor(array &$data, $class, array &$context, \Reflec
      * @param array            $context
      * @param \ReflectionClass $reflectionClass
      * @param array|bool       $allowedAttributes
+     * @param string|null      $format
      *
      * @return object
      *
      * @throws RuntimeException
      */
-    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes)
+    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes, $format = null)
```

Comment:
Sorry I realized that this is BC break... It should be removed. But it's annoying because some normalizers may depend of the format (normalizers of API Platform for instance).

An alternative approach: 

* Rename this method `instantiateComplexObject` (with the new parameter, it can be added before `$context`)
* Add back `instantiateObject` with the current signature and call `instantiateComplexObject` inside it with format as `null`
* Deprecate `instantiateObject`

WDYT?